### DISCUSSION TITLE
Need help understanding the documentation (self-hosting)

### DISCUSSION THREAD

**Discussion by gnacoding** [zitadel/zitadel#7193] [Q&A]
Hi all 

TL;DR: I want to deploy zitadel on a separate server to authenticate applications which either support OIDC or have no authentication mechanism and need forwardAuth. I need help understanding which documentation example is the one I should use.

Backstory: 
I recently tried setting up authentik as my identity provider but due to some bugs and/or unclear documentation which affected my setup I was a bit worried about the long term reliability. Also the developer mentioned that he’s slowly transitioning away form python which could introduce more breaking changes.
That’s why I’m now looking at zitadel

My goal is to run zitadel on a standalone VPS behind traefik. I’d like to use OIDC for portainer, gitlab etc. but also use forwardAuth for apps which don’t have native authentication methods built-in. All applications won’t be on the same VPS as zitadel.

Now, I’m trying to make sense of the official documentation. There are 2 examples posted there:
1. A ZITADEL Load Balancing Example (https://zitadel.com/docs/self-hosting/deploy/loadbalancing-example) 
2. Configure ZITADEL with Traefik (https://zitadel.com/docs/self-hosting/manage/reverseproxy/traefik) 

The first thing I try to understand are the required SSL/TLS certificates for both zitadel and cockroachDB. 
By defining “—tlsmode external” I can provide a Certresolver in traefik for the zitadel-container which will be exposed to the internet. 
But what about cockroachDB? Since cockroach will be an internal container and won’t be exposed to the internet is it expected to have some sort of internal CA to provide a secure certificate? Or do I misunderstand the documentation, and in my scenario a certificate for cockroach is not required (unlikely)?

The second thing I’m trying to understand is the use case for each example.
“A ZITADEL Load Balancing Example” seems to implement forwardAuth by defining rules in the traefik.yaml

```yaml
routers:
    # Redirect HTTP to HTTPS
    router0:
      entryPoints:
      - web
      middlewares:
      - redirect-to-https
      rule: 'HostRegexp(`my.domain`, `{subdomain:[a-z]+}.my.domain`)'
      service: zitadel
    # The actual ZITADEL router
    router1:
      entryPoints:
      - websecure
      service: zitadel
      middlewares:
      - zitadel
      rule: 'HostRegexp(`my.domain`, `{subdomain:[a-z]+}.my.domain`)'
      tls:
        domains:
          - main: "my.domain"
            sans:
              - "*.my.domain"
              - "my.domain"
```

If I understand correctly this is meant to be used in conjunction when additional applications are running on the same VPS.

The example “Configure ZITADEL with Traefik”  seems more in line what I actually want to achieve but I don’t see an example for the traefik.yml which shows a middleware which can be set on a application on other servers to forwardAuth to zitadel.

Can someone please shine some light on this. Thank you!

**Comment by hifabienne** [comment]
@eliobischof @stebenz Can you help here?
